### PR TITLE
Adding DisableCatchAllCheck() switch

### LINKED ISSUE #75: Making catchAll detection optional
The catchAll detection relies on checking the response to a randomly generated email. Effectively it doubles the number of checks, which for batch processing is a problem, as it increases the chance of being blacklisted. 

So it would be nice to perform catchAll detection once and then run the entire batch without it. For that it would be nice to have `DisableCatchAllCheck()` functional switch.

PR follows

### PR DESCRIPTION
**Pull request by lguminski**
closes #75

### CODE REVIEW

**Comment by lryong** [comment]
Hi @lguminski 
Since the `catchAll` check actually requires the `smtpCheck` switch to be turned on, your PR implementation is more like a sub-switch.
Can you do some optimization? For example, by placing a cache for the check results, you can avoid duplicate catchAll checks for the domain multiple times

**Comment by git-hulk** [comment]
Hi @lryong, I think it makes a lot of sense to allow disabling the catch-all feature if it's NOT needed.

**Comment by lryong** [comment]
OK， I would make the CI lint pass

**Comment by lryong** [comment]
@lguminski Please pull down the latest code, I've updated the test cases to make sure the CI passes

File: README.md [hunk 1]
```diff
@@ -105,6 +105,8 @@ func main() {
 }
 ```
 
+> Note: use `verifier.DisableCatchAllCheck()` to disable catchAll check
```
Comment (lryong) [inline README.md:108]: It would be more appropriate to put this comment on line 112， and remove the "Note"

**Comment by lguminski** [comment]
@lryong thanks for fixing tests. Everything looks now good to me

**Review by lryong — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Pull request merged**